### ISSUE TITLE
clearInterval at unmount?

### ISSUE THREAD

**Issue by webdif** [jstejada/react-typist#8]
I get this error when I navigate from a page where there is a Typist instance. It seems that Typist want to modify state although the component is unmounted:

```
warning.js:48 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the undefined component.warning
@ warning.js:48 getInternalInstanceReadyForUpdate
@ ReactUpdateQueue.js:49 enqueueSetState
@ ReactUpdateQueue.js:196 ReactComponent.setState
@ ReactComponent.js:71 (anonymous function)
@ Typist.js:182 adv
@ Typist.js:376 adv
@ Typist.js:372
```

I don't know if I got this right, but maybe it has to do with the setTimeout in `eachRndTimeout` of `utils.js`? For a fix, we can store the timeout id in a variable in `utils.js`, add a function to `clearTimeout` this id, and call this function in the `componentWillUnmount` of `Typist` component?

**Comment by devdoomari** [comment]
#9 solves this issue by manually storing each setTimeouts in an array and running a clearInterval()s  on componentWillUnmount.
Could you provide some feedback on the pull req.?
thanks : )

**Comment by webdif** [comment]
Seems like a good fix for this problem, thank you!
Is #9 will be merged soon and published on npm?

**Comment by devdoomari** [comment]
@webdif 
nope - #9 shouldn't be merged as @jstejada says.
#9 did more than it should in one pull request as I had to do a quick-fix for my company.
(IE fix, babel6, clearinterval)

I think @webdif you could separate out #9 into multiple pull requests and submit them as yours.

**Comment by kendagriff** [comment]
v1.0.2 solved this for me.

**Comment by kendagriff** [comment]
I take that back:

```
warning.js:45 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the undefined component.
warning	@	warning.js:45
getInternalInstanceReadyForUpdate	@	ReactUpdateQueue.js:34
enqueueSetState	@	ReactUpdateQueue.js:181
ReactComponent.setState	@	ReactComponent.js:65
Typist._this.onTypingDone	@	Typist.js:94
(anonymous function)	@	Typist.js:173
```

This appears to be the offending line piece of code in Typist.jsx:

```
  onTypingDone = () => {
    this.setState({ isDone: true });
    this.props.onTypingDone();
  }
```

**Comment by jstejada** [comment]
hey @kendagriff thanks for bringing this up! this should be fixed in https://github.com/jstejada/react-typist/commit/f50798d7fa62114ca4dd89937514efebd78594fc

released 1.0.3 today

**Comment by kendagriff** [comment]
Fixed, thank you!
